### PR TITLE
Refactor gatekeeper endpoint to use DRF

### PR DESCRIPTION
**Pull request by frwickst**
The functionality was kept as close to the original as possible, the main reason for this was a code cleanup, move to DRF as well as me learning how the code works 😄 

Ping @mikkokeskinen (I have to do it this way until you have accepted the invitation to the repo, then I can add you as a reviewer)

### CODE REVIEW

File: backend/gatekeeper/views.py [hunk 1]
```diff
@@ -11,65 +11,79 @@
...
-        header_name = header[5:]
-        if header_name not in ['ACCEPT', 'EXPECT', 'USER_AGENT']:
-            continue
+    def get(self, request):
+        self.sts_arguments['params'] = {k: v for k, v in request.query_params.items() if k.startswith('$')}
+        return self.handle_request(request)
 
-        request_arguments['headers'][header_name] = request.META[header]
+    def post(self, request):
+        print("Data: \n", self.request.data)
```
Comment (mikkokeskinen) [inline backend/gatekeeper/views.py]: Maybe remove debug print

File: backend/gatekeeper/views.py [hunk 2]
```diff
@@ -11,65 +11,79 @@
...
+        self.sts_arguments['json'] = self.request.data
+        return self.handle_request(request)
 
-    # TODO: access control
-    if request.method == 'GET':
-        # TODO: validate GET parameters
```
Comment (mikkokeskinen) [inline backend/gatekeeper/views.py]: Did you gather the removed TODOs somewhere?

File: backend/gatekeeper/views.py [hunk 3]
```diff
@@ -123,48 +135,68 @@ def index(request, path):
...
+                response[name] = value
 
-    # If the content is JSON fix the urls to point to this view, not directly to the STS
-    # (Filter 404 out because FraunhoferIOSB/SensorThingsServer will return 404 with the
-    # content type application/json with the content "Nothing found".)
-    if cgi.parse_header(content_type)[0] == 'application/json' and r.status_code != 404:
-        data = json.loads(content, object_pairs_hook=OrderedDict, encoding=r.encoding)
+        return response
 
+    def remap_response_content_urls(self, data):
```
Comment (mikkokeskinen) [inline backend/gatekeeper/views.py]: These URL rewrites can be removed as we changed the URL in the STS settings

File: backend/gatekeeper/views.py [hunk 4]
```diff
@@ -11,65 +11,79 @@
...
-from gatekeeper.models import Datastream, Thing
-from gatekeeper.utils import parse_sta_url
+from .models import Datastream, Thing
+from .utils import parse_sta_url
 
 
-@csrf_exempt
-def index(request, path):
-    """This is a quick and dirty proof of concept for proxying the SensorThings API queries to the real STA server.
+class Gatekeeper(APIView):
```
Comment (mikkokeskinen) [inline backend/gatekeeper/views.py]: How does APIView handle OPTIONS? I think it should be forwarded to the STS too.

Comment (frwickst) [reply backend/gatekeeper/views.py]: Added

File: backend/gatekeeper/views.py (hunk 1 above)
Comment (frwickst) [reply backend/gatekeeper/views.py]: Fixed

File: backend/gatekeeper/views.py (hunk 2 above)
Comment (frwickst) [reply backend/gatekeeper/views.py]: Added access control todo to top of class. The validated GET parameters, hmm, I don't know about that one, should be actually do that? Isn't that something that the STS will check, as you said yourself, we should resemble a proxy as much as possibly.

File: backend/gatekeeper/views.py (hunk 3 above)
Comment (frwickst) [reply backend/gatekeeper/views.py]: Removed

File: backend/gatekeeper/views.py [hunk 5]
```diff
@@ -71,6 +76,14 @@ def delete(self, request):
         self.sts_arguments['params'] = {k: v for k, v in request.query_params.items() if k.startswith('$')}
         return self.handle_request(request)
 
+    def options(self, request, *args, **kwargs):
+        self.sts_arguments['params'] = {k: v for k, v in request.query_params.items() if k.startswith('$')}
```
Comment (mikkokeskinen) [inline backend/gatekeeper/views.py]: Now that this is in three places, could it be a method "get STA parameters" or something?

Comment (frwickst) [reply backend/gatekeeper/views.py]: Sure, that is a good point!

Comment (frwickst) [reply backend/gatekeeper/views.py]: I'm wondering, maybe these should always be set. Any query parameter should always be passed to the STS right? No matter the HTTP method, be it POST or GET for instance.

Comment (frwickst) [reply backend/gatekeeper/views.py]: Done

**Pull request merged**